realm/src/androidTest/java/io/realm/RealmInMemoryTest.java

```diff
@@ -0,0 +1,215 @@
+            fail("Realm.getInMemoryInstance should fail with illegal argument");
+        } catch (IllegalArgumentException iae) { //NOPMD
+        }
+
+        realmOnDisk.close();
+    }
+
+    public void testDelete() {
+        RealmConfiguration configuration = testRealm.getConfiguration();
+        try {
+            Realm.deleteRealm(configuration);
+            fail("Realm.deleteRealm should fail with illegal state");
+        } catch (IllegalStateException iae) { //NOPMD
+        }
+
+        // Nothing should happen when delete a closed in-mem-realm.
+        testRealm.close();
+        testRealm = null;
+        Realm.deleteRealm(configuration);
+    }
+
+    public void testWriteCopyTo() throws IOException {
+        String fileName = IDENTIFIER + ".realm";
+        String encFileName = IDENTIFIER + ".enc.realm";
+        byte[] key = TestHelper.getRandomKey();
+
+        Realm.deleteRealm(new RealmConfiguration.Builder(getContext()).name(fileName).build());
+        Realm.deleteRealm(new RealmConfiguration.Builder(getContext()).name(encFileName).build());
+
+        testRealm.beginTransaction();
+        Dog dog = testRealm.createObject(Dog.class);
+        dog.setName("DinoDog");
+        testRealm.commitTransaction();
+
+        // Test a normal realm file
+        testRealm.writeCopyTo(new File(getContext().getFilesDir(), fileName));
+        Realm onDiskRealm = Realm.getInstance(new RealmConfiguration.Builder(getContext()).name(fileName).build());
+        assertEquals(onDiskRealm.allObjects(Dog.class).size(), 1);
+        onDiskRealm.close();
+
+        // Test a encrypted realm file
+        testRealm.writeEncryptedCopyTo(new File(getContext().getFilesDir(), encFileName), key);
+        onDiskRealm = Realm.getInstance(new RealmConfiguration.Builder(getContext()).name(encFileName)
+                .encryptionKey(key).build());
+        assertEquals(onDiskRealm.allObjects(Dog.class).size(), 1);
+        onDiskRealm.close();
+        // Test with a wrong key
+        try {
+            Realm.getInstance(new RealmConfiguration.Builder(getContext()).name(encFileName)
+                    .encryptionKey(TestHelper.getRandomKey(42)).build());
+            fail("Realm.getInstance should fail with illegal argument");
+        } catch (IllegalArgumentException iae) { //NOPMD
+        }
+    }
+
+    public void testCreateWithConfiguration() {
+        testRealm.beginTransaction();
+        Dog dog = testRealm.createObject(Dog.class);
+        dog.setName("DinoDog");
+        testRealm.commitTransaction();
+
+        RealmConfiguration configuration = new RealmConfiguration.Builder(getContext())
+                .name(IDENTIFIER).inMemory().build();
+        Realm realm = Realm.getInstance(configuration);
+
+        assertEquals(realm.allObjects(Dog.class).first().getName(), "DinoDog");
+        realm.close();
+    }
+
+    public void testMultiThread() throws InterruptedException, ExecutionException {
```

Comment:
Add a comment about what this is testing. From what I can tell it is that a in-memory realm is not fully deleted until all threads has released their instance.